### PR TITLE
Refactor queries related stuff

### PR DESCRIPTION
**Pull request by kanitw**
(prepare for related views)

### CODE REVIEW

File: src/queries/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {Query} from 'compassql/build/src/query/query';
+import {Store} from 'redux';
+import {StateWithHistory} from 'redux-undo';
+import {StateBase} from '../../build/src/models/index';
+import {resultRequest} from '../actions/result';
+
+export function makeQueries(store: Store<StateWithHistory<Readonly<StateBase>>>, query: Query) {
```
Comment (vlandham) [inline src/queries/index.ts]: the verb 'make' is consistent with other areas in the code where we dispatch ? or is this the first time this is happening?

File: src/models/index.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export const DEFAULT_STATE: StateBase = {
   config: DEFAULT_VOYAGER_CONFIG,
   dataset: DEFAULT_DATASET,
   shelf: DEFAULT_SHELF_SPEC,
-  result: DEFAULT_RESULT,
+  result: DEFAULT_RESULT_INDEX,
```
Comment (starry97) [inline src/models/index.ts]: I'm not sure why you changed the name to DEFAULT_RESULT_INDEX. It looks like we are using ResultIndex in other places too, but it seems like a confusing name to me.

Comment (kanitw) [inline src/models/index.ts]: `state.result` stores an dictionary (index) for different results.    

`result.main` is for the main result, but we will have other results too.

File: src/queries/index.ts (hunk 1 above)
Comment (kanitw) [inline src/queries/index.ts]: I'll name it `dispatchQueries` then.

**Comment by kanitw** [comment]
I'm making this depends on https://github.com/vega/voyager/pull/476.

**Pull request merged**